41-vaccine-temp-monitor/firmware/arduino/src/main.cpp

```diff
@@ -0,0 +1,528 @@
+        if (PRODUCT_UID[0])
+        {
+            JAddStringToObject(req, "product", PRODUCT_UID);
+        }
+        JAddStringToObject(req, "mode", "periodic");
+        JAddStringToObject(req, "vinbound", "usb:5;high:120;normal:240;low:480;dead:0");
+        JAddStringToObject(req, "voutbound", "usb:5;high:60;normal:120;low:240;dead:0");
+        notecard.sendRequestWithRetry(req, RETRY_S);
+    }
+
+    // Optimize voltage variable behaviors for LiPo battery
+    if (J *cmd = notecard.newCommand("card.voltage"))
+    {
+        JAddStringToObject(cmd, "mode", "lipo");
+        notecard.sendRequest(cmd);
+    }
+
+    // Ensure the accelerometer has been activated
+    if (J *cmd = notecard.newCommand("card.motion.mode"))
+    {
+        JAddBoolToObject(cmd, "start", true);
+        JAddNumberToObject(cmd, "sensitivity", 1);
+        notecard.sendRequest(cmd);
+    }
+
+    // Configure GPS heartbeat to verify last location reported
+    if (J *cmd = notecard.newCommand("card.location.track"))
+    {
+        JAddBoolToObject(cmd, "start", true);
+        JAddBoolToObject(cmd, "sync", true);
+        JAddBoolToObject(cmd, "heartbeat", true);
+        JAddNumberToObject(cmd, "hours", 12);
+        notecard.sendRequest(cmd);
+    }
+
+    // Establish a template to optimize queue size and data usage
+    if (J *cmd = notecard.newCommand("note.template"))
+    {
+        JAddStringToObject(cmd, "file", "status.qo");
+        if (J *body = JAddObjectToObject(cmd, "body"))
+        {
+            JAddBoolToObject(body, "alert", TBOOL);
+            JAddBoolToObject(body, "low_batt", TBOOL);
+            JAddNumberToObject(body, "temp", TFLOAT16);
+            JAddNumberToObject(body, "voltage", TFLOAT16);
+            notecard.sendRequest(cmd);
+        }
+    }
+}
+
+bool currentlyMoving(void)
+{
+    bool moving;
+    uint32_t movementTimeSecs = 0;
+
+    // Increase motion sensitivity to help detect movement
+    if (J *cmd = notecard.newCommand("card.motion.mode"))
+    {
+        JAddBoolToObject(cmd, "start", true);
+        JAddNumberToObject(cmd, "sensitivity", 5);
+        notecard.sendRequest(cmd);
+    }
+
+    // Determine whether or not the module is moving.
+    J *rsp = _notecard.requestAndResponse(_notecard.newRequest("card.triangulate"));
+    if (!rsp)
+    {
+        moving = false; // prioritize battery life on error
+    }
+    else
+    {
+        uint32_t movementTimeSecs = JGetInt(rsp, "motion");
+        NoteDeleteResponse(rsp);
+
+        // Wait long enough for the accelerometer to record a new timestamp.
+        ::delay(1500);
+
+        rsp = _notecard.requestAndResponse(_notecard.newRequest("card.triangulate"));
+    }
+
+    // Restore motion sensitivity to preserve battery life
+    if (J *cmd = notecard.newCommand("card.motion.mode"))
+    {
+        JAddBoolToObject(cmd, "start", true);
+        JAddNumberToObject(cmd, "sensitivity", 1);
+        notecard.sendRequest(cmd);
+    }
+
+    return valid;

```

Comment:
`valid` not declared. This function seems incomplete.

Reply:
Yeah, I don't know what I pushed up exactly, that became `moving` and subsequently made sense. :shrug: